content/posts/newsletter-013/index.md

```diff
@@ -66,6 +66,90 @@ If needed, a section can be split into subsections with a "------" delimiter.
 
 ## Library & Tooling Updates
 
+### Piston
+
+Piston is a modular game engine written in Rust.
+
+A new [Discord channel](https://t.co/8YOj3auDr9?amp=1) has been set up
+for the Piston project.
+
+Piston consists of a core library "piston" which itself are composed of
+smaller libraries for abstracting input, window and event loop.
+This design helps reducing breaking changes in the ecosystem.
+
+The core library `pistoncore-input` is now stabilized and reached 1.0!
+This is the most important core abstraction, because it glues all
+libraries that are not independent of the core.
+
+------
+
+Dyon is a rusty dynamically typed scripting language.
+It is developed and maintained as part of the Piston project,
+but can be used as a standalone library.
+
+Dyon is designed from the bottom up to be a good gamedev scripting language for Rust.
+It uses a lifetime checker instead of garbage collection, a mutability checker,
+optional namespaces and ad-hoc types, named argument syntax,
+4D vectors and HTML colors, plus a lot more features!
+
+Recently, Dyon got better macro integration for native Rust types
+using `#` as a prefix.
+
+Here is an example of this feature is being tested in
+an experimental offline 3D renderer (not open sourced):
+
+```text
+// Called by `set_simple(scene: _, sdf: _, id: _)`.
+dyon_fn!{fn set_simple__scene_sdf_id(
+    scene: #&mut SimpleScene,
+    sdf: #&Sdf,
+    id: f64
+) {
+    scene.sdfs[id as usize] = sdf.clone()
+}}
+```
+
+To follow updates on Dyon, check out the subreddit
+[/r/dyon](https://old.reddit.com/r/dyon/).
+
+------
+
+Piston-Graphics is a library for 2D graphics, written in Rust,
+that works with multiple backends.
+
+`Stencil::Increment` has been added and the ecosystem
+has been updated to the latest version.
+
+------
+
+The research branch of the Piston project, AdvancedResearch,
+has released a new ECS library [Nano-ECS](https://github.com/advancedresearch/nano_ecs).
+
+This ECS design stores all components in a single array
+and uses bit masks for enabling/disabling components.
+An entity can have maximum 64 components and must be initalized
+with all components it uses in the future.
+Each entity has a slice into the array that stores all components.
+The `World` object, `Component` and systems are generated using macros.
+
+One research project with Nano-ECS is to prototype a UI framework
+for Rust with a UI editor (not open sourced yet).
+This project uses Piston-Graphics by default,
+but can generate draw commands for processing by other 2D APIs.
+It is also possible to override rendering of widgets for
+custom looks with Piston-Graphics, which is often useful in gamedev.
+Recently, this project has gotten to a place where
+[tree-view interaction](https://tinyurl.com/y63f5xkv) is working.
+
+[Piston Discord Channel]: https://t.co/8YOj3auDr9?amp=1
```

Comment:
These [reference-style links](https://spec.commonmark.org/0.29/#reference-link) don't seem to be used anywhere - did you mean to use [inline links](https://spec.commonmark.org/0.29/#inline-link) instead?

Reply:
@17cupsofcoffee Thank you! I hope I got this right this time.

Reply:
This looks better :+1:

These links still aren't used anywhere in the post, though: 
* `[@PistonDeveloper at Twitter]`
* `[Piston]`
* `[Dyon]`
* `[Piston-Graphics]`

If you want those URLs to be accessible, you'll need to either reference those links somewhere in the text or replace them with inline links. Markdown's link syntax is a bit inscrutable, if you get stuck let me know what it is you're trying to achieve and I can help work out the markup for it :+1:

FYI: You can click this icon on the Files Changed page of the PR to preview your changes - that's how I've been checking what the rendered post looks like: 
![image](https://user-images.githubusercontent.com/784533/92154714-237c4e00-ee1e-11ea-855b-3f463a96a382.png)

Reply:
@17cupsofcoffee Thank you! I've fixed the link and resolved the merge conflict.